- Updated odata to sql translator, Translate method to modify queries…

ahesami opened the pull request:
… that have startswith, endswith or contains with lower or upper functions. Added tests to verify changes.

- Updated nuspec file and updated the version.
- Updated readme file with the change.

ahesami committed:
- Updated odata to sql translator, Translate method to modify queries that have startswith, endswith or contains with lower or upper functions. Added tests to verify changes.
- Updated nuspec file and updated the version.
- Updated readme file with the change.

aboo commented on azure-documentdb-odata-sql-samples/ODataToSqlSamples.cs:
```diff
@@ -792,6 +792,37 @@ public void Translate_ReturnsCorrectResult_WhenANestedRangeVariableIsPresent()
+		[DataRow("http://localhost/User?$filter=startswith(toupper(englishName), 'microsoft')", "SELECT * FROM c WHERE STARTSWITH(c.englishName,'microsoft', true) ")]
+		[DataRow("http://localhost/User?$filter=contains(toupper(englishName), 'microsoft')", "SELECT * FROM c WHERE CONTAINS(c.englishName,'microsoft', true) ")]
+		[DataRow("http://localhost/User?$filter=endswith(toupper(englishName), 'microsoft')", "SELECT * FROM c WHERE ENDSWITH(c.englishName,'microsoft', true) ")]
+		public void Translate_ReturnsCaseInsensitiveStringFunction_WhenODataQueryContainsCaseInsensitiveStringQuery(string uriString, string expectedSqlQuery)
```
The test fails for following more complex scenarios

```csharp
[DataRow("http://localhost/User?$filter=endswith(toupper(englishName), 'microsoft') and startswith(tolower(englishName),'random')", "SELECT * FROM c WHERE ENDSWITH(c.englishName,'microsoft', true) AND STARTSWITH(c.englishName,'random', true)")]
[DataRow("http://localhost/User?$filter=endswith(toupper(englishName), 'microsoft') or startswith(englishName,'randoM')", "SELECT * FROM c WHERE ENDSWITH(c.englishName,'microsoft', true) OR STARTSWITH(c.englishName,'randoM')")]
[DataRow("http://localhost/User?$filter=endswith(toupper(englishName), 'microsoft') and endswith(tolower(englishName),'random')", "SELECT * FROM c WHERE ENDSWITH(c.englishName,'microsoft', true) AND ENDSWITH(c.englishName,'random', true)")]
		
```

ahesami replied:
Addressed

aboo commented on README.md:
```diff
@@ -21,6 +21,7 @@ PRs are welcome.
 Converts [OData V4](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html) queries to [DocumentDB SQL](https://azure.microsoft.com/en-us/documentation/articles/documentdb-sql-query/) queries. 
 
 ## Release Notes
+* 3.1.80 Added support for optimized case-insensitive string queries
```
please mention that this is a breaking change and tolower and toupper will no longer perform what they are expecting to do.

ahesami replied:
Addressed

ahesami committed:
Updated odata to sql translator

- Updated odata to sql translator and fixed an issue regarding complex queries.
- Updated readme file and added breaking changes.
